pkg/watcher/metricsprovider.go

```diff
@@ -60,4 +64,6 @@ type MetricsProviderOpts struct {
 	Name      string
 	Address   string
 	AuthToken string
+	InsecureSkipVerify	bool
+	EnableOpenShiftAuth	bool
```

Comment:
@zorro786 , these changes are particularly necessary for environments like AWS and GCP as these envs do not allow InsecureSkipVerify by default. Env is not really friendly to users who install trimarans from operators, etc. 
And the goal is to make upstream changes in scheduler plugins as well so developers are aware of environment differences for different Cloud providers.

Reply:
@wangchen615 I am fine with adding "InsecureSkipVerify" option here, as it may be common for all metric providers.
My main concern is adding "EnableOpenShiftAuth" as a param here. There should be no platform specific parameters here, only generic applicable to all. Lookup how I solved similar additional config issues with `signalfx.go` for `signalFxHostNameSuffixKey` to not add that param over here.

Reply:
@zorro786 I see. Thanks for the suggestion. Let me discuss this with our team and come back to you.